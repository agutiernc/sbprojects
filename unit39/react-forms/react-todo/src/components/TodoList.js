import React, { useState } from 'react'
import Todo from './Todo'
import NewTodoForm from './NewTodoForm'

const TodoList = () => {
  const [todos, setTodos] = useState([])

  const newTodo = (todoObj) => setTodos(todos => [...todos, todoObj])
  const deleteTodo = (id) => setTodos(todos => todos.filter(t => t.id !== id))

  return (
    <div>
      <h1>Todos</h1>

      <div>
        <NewTodoForm newTodo={newTodo} />
      </div>

      <div>
        <ul>
          {
            todos.map(t => (
              <Todo
                key={t.id}
                id={t.id}
                todo={t.todo}
                handleRemove={deleteTodo}
              />
            ))
          }
        </ul>
      </div>
    </div>
  )
}

export default TodoList