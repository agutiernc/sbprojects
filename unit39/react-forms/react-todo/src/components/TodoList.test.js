import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import TodoList from "./TodoList";

const newTodo = (todo = 'test this todo') => {
  const todoInput = screen.getByLabelText('Add todo:')

  fireEvent.change(todoInput, { target: { value: todo } })

  const btn = screen.getByText('Submit')

  fireEvent.click(btn)
}


// smoke test
it('renders without crashing', () => {
  render(<TodoList />)
})

// snapshot test
it('matches snapshot', () => {
  const { asFragment } = render(<TodoList />)

  expect(asFragment()).toMatchSnapshot()
})

it('can add a new todo', () => {
  render(<TodoList />)

  expect(screen.queryByText('x')).not.toBeInTheDocument()

  newTodo()

  expect(screen.getByText('x')).toBeInTheDocument()
  expect(screen.getByText('test this todo -')).toBeInTheDocument()
})

it('can delete a todo', () => {
  render(<TodoList />)

  newTodo()

  fireEvent.click(screen.getByText('x')) // click delete button

  // todo is removed
  expect(screen.queryByText('test this todo -')).not.toBeInTheDocument()
})