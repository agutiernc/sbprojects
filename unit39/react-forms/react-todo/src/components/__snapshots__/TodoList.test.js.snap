// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Todos
    </h1>
    <div>
      <div>
        <form>
          <label
            for="todo"
          >
            Add todo:
          </label>
          <input
            id="todo"
            name="todo"
            type="text"
            value=""
          />
          <button>
            Submit
          </button>
        </form>
      </div>
    </div>
    <div>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
